### PR TITLE
Fix: Fixed issue where the context menu would sometimes stay open

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
The context menu does not close when executing a command in a submenu. This pr fixes this.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Before

https://user-images.githubusercontent.com/46631671/201942336-8f8e8b2e-6f8f-41dd-b489-ea64ba8a1c2c.mp4

After

https://user-images.githubusercontent.com/46631671/201942330-027bb914-e128-4af0-b895-d9d03a06f798.mp4

### CODE REVIEW

File: src/Files.App/BaseLayout.cs [hunk 1]
```diff
@@ -587,14 +596,23 @@ private async Task LoadMenuItemsAsync()
...
+			primaryElements
+				.OfType<AppBarButton>()
+				.ForEach(button => button.Click += closeHandler);
+			secondaryElements
+				.OfType<AppBarButton>()
+				.Select(item => item.Flyout)
+				.OfType<MenuFlyout>()
+				.SelectMany(menu => menu.Items)
+				.OfType<MenuFlyoutItem>()
+				.ForEach(button => button.Click += closeHandler);
```
Comment (QuaintMako) [inline src/Files.App/BaseLayout.cs:614]: Those lines are a repetition of l.541 to l.553.
Would it be possible to fuse them inside a method?

Comment (cinqmilleans) [reply src/Files.App/BaseLayout.cs]: It's done.

File: src/Files.App/BaseLayout.cs [hunk 2]
```diff
@@ -606,6 +600,24 @@ private async Task LoadMenuItemsAsync()
 			}
 		}
 
+		private void AddCloseHandler(IList<ICommandBarElement> primaryElements, IList<ICommandBarElement> secondaryElements)
+		{
+			// Workaround for WinUI (#5508)
```
Comment (yaira2) [inline src/Files.App/BaseLayout.cs:605]: Can you include a link to the issue?

Comment (cinqmilleans) [reply src/Files.App/BaseLayout.cs:605]: The comment comes from a previous fix for primary elements. This pr fixes the submenus of the secondary elements. Is it worth keeping this comment? What link should I put?

Comment (yaira2) [reply src/Files.App/BaseLayout.cs:605]: If it's still a workaround, we should link to the issue on WinUI

**Pull request merged**